samcli/lib/telemetry/metric.py

```diff
@@ -29,16 +30,120 @@
 No side effect will result in this as it is write-only for code outside of telemetry.
 Decorators should be used to minimize logic involving telemetry.
 """
-_METRICS = dict()
+_METRICS: Optional[defaultdict] = None
+
+
+class MetricName(Enum):
+    # Metrics that can be reported only once in a session
+    installed = ("installed", True)
+    commandRun = ("commandRun", True)
+
+    # Metrics that can be reported any number of times during a session
+    # e.g. docker_image_download_status
+    templateWarning = ("templateWarning", False)
```

Comment:
is the idea to continuously track certain things throughout the session, I'm assuming these are no longer anything to do with the command that was used to run SAM CLI then, since those options and flags are static once they are invoked.

Reply:
Options and flags are static and can be accessed via the context object. But sometimes we want to track things that are created within the command actions. One example is for `runtimeMetric`, we want to track the runtime used for functions during a `local invoke` run.